src/builder/dataflow.rs

```diff
@@ -82,7 +82,8 @@ impl<'f> Dataflow for DFGBuilder<'f> {
 }
 
 /// Wrapper around [`DFGBuilder`] used to build other dataflow regions.
-pub struct DFGWrapper<'b, T>(DFGBuilder<'b>, PhantomData<T>);
+// Stores option of DFGBuilder so it can be taken out without moving.
```

Comment:
Is it possible to have a wrapper with the builder taken out? I only see that being done in `finish()`, which immediately drops the wrapper.

Reply:
this was a hack to make it possible to call "finish" on the inside builder (I wasn't allowed to move it out of the wrapper to call finish because it implements Drop - this may be a more serious)

Reply:
Fair.
I think the only way to get around that would be some `MaybeUninit` and `ManuallyDrop` trickeries, but I wouldn't do that unless there's a perf problem.